### DISCUSSION TITLE
Pocketbase in a Kubernetes pod not accessable with Pocketbase JS SDK

### DISCUSSION THREAD

**Discussion by jorishaenseler15** [pocketbase/pocketbase#905] [Q&A]
Hello

**Problem**
I am trying to host my pocketbase instance in a kubernetes pod but the pocketbase js sdk wich is in another pod is not able to access this instance. In the console is the following error:

<img width="704" alt="Screenshot 2022-11-02 at 23 04 12" src="https://user-images.githubusercontent.com/87999120/199612577-28be0372-ca5a-48b6-a2f1-82374bc4add2.png">

But the strange thing is that when I go into the pod where the app with the pocketbase js sdk runs I can call the pocketbase api with curl and I get a response

Example curl:
`curl http://pocketbase:8090/api` 

And this request also returns the html
`curl http://pocketbase:8090/_/`

**My Dockerfile for pocketbase**
```
FROM alpine:3 as downloader

ARG TARGETOS
ARG TARGETARCH
ARG TARGETVARIANT
ARG VERSION=0.7.9

ENV BUILDX_ARCH="${TARGETOS:-linux}_${TARGETARCH:-amd64}${TARGETVARIANT}"

RUN wget https://github.com/pocketbase/pocketbase/releases/download/v${VERSION}/pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && unzip pocketbase_${VERSION}_${BUILDX_ARCH}.zip \
    && chmod +x /pocketbase

FROM alpine:3
RUN apk update && apk add ca-certificates && rm -rf /var/cache/apk/*

EXPOSE 8090

COPY --from=downloader /pocketbase /usr/local/bin/pocketbase
ENTRYPOINT ["/usr/local/bin/pocketbase", "serve", "--http=0.0.0.0:8090", "--dir=/pb_data", "--publicDir=/pb_public"]
```

### The following kubernetes objects are all in the same namespace

**My service.yaml**
```apiVersion: v1
kind: Service
metadata:
  annotations:
    kompose.version: 1.26.1 (HEAD)
  creationTimestamp: null
  labels:
    io.kompose.service: pocketbase
  name: pocketbase
spec:
  ports:
    - name: "8090"
      port: 8090
      targetPort: 8090
  selector:
    io.kompose.service: pocketbase
status:
  loadBalancer: {}
```

**My deployment.yaml**

```apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kompose.cmd: kompose convert -f docker_teslarental/docker-compose.yaml
    kompose.version: 1.26.1 (HEAD)
  creationTimestamp: null
  labels:
    io.kompose.service: pocketbase
  name: pocketbase
spec:
  replicas: 1
  selector:
    matchLabels:
      io.kompose.service: pocketbase
  strategy:
    type: Recreate
  template:
    metadata:
      annotations:
        kompose.version: 1.26.1 (HEAD)
      creationTimestamp: null
      labels:
        io.kompose.service: pocketbase
    spec:
      containers:
        - image: dockerhubname/pocketbase:1.0
          name: pocketbase
          imagePullPolicy: Always
          ports:
            - containerPort: 8090
          resources: {}
          volumeMounts:
            - mountPath: /pb_data
              name: pocketbase-claim0
      restartPolicy: Always
      volumes:
        - name: pocketbase-claim0
          persistentVolumeClaim:
            claimName: pocketbase-claim0
status: {}
```

Thanks for help!!!!:)

**Comment by ganigeorgiev** [comment]
If the above error is happening from a client-side request in the browser and is not forwarded from a SSR action, then I'm not sure what could be the reason for the failure.

But if you are using Node 17+, then it may be related to https://github.com/pocketbase/js-sdk/issues/21 due to the different dns resolution compared to the Go dns resolver.

You could try changing the entrypoint to use the ipv6 equivalent of `0.0.0.0`, aka. `--http=[::]:8090`

**Reply by jorishaenseler15** [reply]
Thanks for the answer

It did not change anything when I changed to ipv6 `--http=[::]:8090`

And I also tried to use a node image which is using node 16 for my application (then I used `--http=127.0.0.1:8090`) with the pocketbase js sdk. But the same error still occurs.

I now hosted the pocketbase instance outside of my cluster and it worked. But if somebody still has an idea it would be nice to hear!

**Comment by pchasco** [comment]
It seems way overkill to host PB in k8s because you can only scale it to one instance, and you need to use a local volume on the filesystem for SQLite.
